elkodon_bb/posix/tests/barrier_tests.rs

```diff
@@ -28,9 +28,11 @@ fn barrier_blocks() -> Result<(), BarrierCreationError> {
         });
 
         sut.wait();
-        assert_that!(counter.load(Ordering::Relaxed), eq 0);
```

Comment:
Yes, if the counter would be here != 0 the assert would prevent that the following two `wait()` would not be reached and the threads that are running right above this could would deadlock.